src/components/Search/Search.tsx

```diff
@@ -258,7 +273,7 @@ export const Search = ({
                         aria-label="clear-search"
                         style={{ margin: '1rem', marginLeft: 'auto', cursor: 'pointer', display: 'flex' }}
                         onClick={() => {
-                            setValue('');
+                            isOnControl ? setPropsValue?.('') : setStateValue('');
```

Comment:
Maybe. instead of doing this ternary on every step, we need to update the value we can have a `updateValue()` that allows us to have the change of the value in a single place.

```
const updateValue = (value) => {
    propsValue !== undefined ? setPropsValue?.('') : setStateValue('')
}
```